src/cdk/datepicker/calendar-view.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Subject} from 'rxjs';
+
+/**
+ * An abstract calendar that is used as part of the datepicker. This abstract calendar class
+ * contains all necessary parts needed for a generic datepicker component.
+ */
+export abstract class CalendarView<D> {
+
+  /** The date representing when to start the calendar. */
+  abstract activeDate: D;
+
+  /** The minimum selectable date. */
+  abstract minDate: D | null;
+
+  /** The maximum selectable date. */
+  abstract maxDate: D | null;
+
+  /** The currently selected date. */
+  abstract selected: D | null;
+
+  /**
+   * Emits whenever there is a state change that needs to be responded to.
+   */
+  readonly changes = new Subject<D>();
```

Comment:
What kind of changes do we expect here? 
Are we going to include `selectedChange` and `activeDateChange`?

Reply:
I think we should have two state changes, one for `selectedChange` and another one for `activeDateChange` so that we can tell which date has changed. @mmalerba is this what you were thinking?

Reply:
I'll be adding a selectedChange emitter to CalendarView for when the selected date changes in the views. If anything else comes up that the datepicker needs to react to, I will add in the future.